test/validate/linux_validate.go

```diff
@@ -158,6 +225,31 @@ func cnsStateFileIps(result []byte) (map[string]string, error) {
 	return cnsPodIps, nil
 }
 
+func azureCNSNonCiliumStateFileIPs(result []byte) (map[string]string, error) {
+	var azureCNSNonCiliumResult AzureCniState
+	err := json.Unmarshal(result, &azureCNSNonCiliumResult)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal azure cni endpoint list")
+	}
+
+	azureCnsPodIps := make(map[string]string)
+	for _, v := range azureCNSNonCiliumResult.AzureCniState.ExternalInterfaces {
+		for _, networks := range v.Networks {
+			for _, ip := range networks.Endpoints {
+				pod := ip.PODName
+				ipv4 := ip.IPAddresses[0].IP
+				azureCnsPodIps[ipv4] = pod
```

Comment:
I think this would pass in both cases, but want to make sure the variable names though. In case of dual stack the first IP address is IPv4, i thought it would be IPv6 ?

Reply:
yes here we need to enhance because in dual nic scenario. two ips may be all ipv4 addresses